src/Illuminate/Database/Connection.php

```diff
@@ -14,15 +14,14 @@
 use Illuminate\Database\Events\TransactionRolledBack;
 use Illuminate\Database\Query\Builder as QueryBuilder;
 use Illuminate\Database\Query\Expression;
-use Illuminate\Database\Query\Grammars\Grammar as QueryGrammar;
 use Illuminate\Database\Query\Processors\Processor;
 use Illuminate\Database\Schema\Builder as SchemaBuilder;
 use Illuminate\Support\Arr;
 use LogicException;
 use PDO;
 use PDOStatement;
 
-class Connection implements ConnectionInterface
+abstract class Connection implements ConnectionInterface
```

Comment:
Was converting this to an abstract class and the related changes necessary to this PR? If not, it just adds a bit of noise to the PR.

Reply:
The `Connection` class has a weird (?) implementation about the schema and query grammar: The connection class was returning a default query grammar in `getDefaultQueryGrammar` which is overridden in _all_ database connection classes with specific grammar (making it useless?). The `getDefaultSchemaGrammar` is not returning anything but is overridden again by everyone.

The `Connection` class has been declared `abstract` to no longer provide implementations for `getDefaultQueryGrammar` and `getDefaultSchemaGrammar`. This may sound unnecessary at first, but as the `Grammar` class is now abstract because the `getDriverName` method needs to be implemented, a default `Grammar` object can no longer be created, so the change is required.